### PR TITLE
[Checkpoint v6] Check part file exist

### PR DESCRIPTION
**Pull request by zhangchiqing**
Addressing [this comment](https://github.com/onflow/flow-go/pull/3273/files#r984071121)

### CODE REVIEW

File: ledger/complete/wal/checkpoint_v6_writer.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 )
 
 const subtrieLevel = 4
-const subtrieCount = 1 << subtrieLevel
+const subtrieCount = 1 << subtrieLevel // 16
```
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply ledger/complete/wal/checkpoint_v6_writer.go:20]: not sure what do you mean. The comment `16` here means the result is `16`

[the remark this answers is not in the document]
Comment (Tonix517) [reply ledger/complete/wal/checkpoint_v6_writer.go:20]: nvm - somehow my brain treated `//` as a division..

[the remark this answers is not in the document]
Comment (zhangchiqing) [reply ledger/complete/wal/checkpoint_v6_writer.go:20]: maybe was too much python? :)

[the remark this answers is not in the document]
Comment (Tonix517) [reply ledger/complete/wal/checkpoint_v6_writer.go:20]: yea unfortunately 🤦

File: ledger/complete/wal/checkpoint_v6_reader.go [hunk 2]
```diff
@@ -82,6 +93,78 @@ func readHeader(filePath string) ([]uint32, uint32, error) {
...
 // }
 
+// allPartFileExist check if all the part files of the checkpoint file exist
+// it returns nil if all files exist
+// it returns os.ErrNotExist if some file is missing, use (os.IsNotExist to check)
+// it returns err if running into any exception
+func allPartFileExist(dir string, fileName string, totalSubtrieFiles int) error {
+	// check all subtrie part file exist
+	for i := 0; i < totalSubtrieFiles; i++ {
+		filePath, _, err := filePathSubTries(dir, fileName, i)
```
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply ledger/complete/wal/checkpoint_v6_reader.go:103]: yeah, [it's changed here](https://github.com/onflow/flow-go/pull/3319/files#diff-2ff6507ee260a1e37156b4cae4cc259e7688bbd0007993b20fa193824073105aR58)

File: ledger/complete/wal/checkpoint_v6_reader.go [hunk 3]
```diff
@@ -82,6 +93,78 @@ func readHeader(filePath string) ([]uint32, uint32, error) {
...
+	return nil
+}
+
+var errCheckpointFileExist = errors.New("checkpoint file exists already")
+
+// noneCheckpointFileExist check if none of the checkpoint header file or the part files exist
+// it returns nil if none exists
+// it returns errCheckpointFileExist if a checkpoint file exists already
+// it returns err if running into any other exception
+func noneCheckpointFileExist(dir string, fileName string, totalSubtrieFiles int) error {
```
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply ledger/complete/wal/checkpoint_v6_reader.go:131]: yeah I wonder what pattern could match all the files?

If the file is `root.checkpoint` then, we want to only match:
```
root.checkpoint
root.checkpoint.000
root.checkpoint.001
root.checkpoint.002
...
root.checkpoint.014
root.checkpoint.015
root.checkpoint.016
```

But not `root.checkpoint.017`, `root.checkpoint.bak`, or `root.checkpoint.bak.000`

[the remark this answers is not in the document]
Comment (m4ksio) [reply ledger/complete/wal/checkpoint_v6_reader.go:131]: You can use `root.checkpoint.0-90-90-9` and then filter out numbers, it should still be faster

**Review by m4ksio — APPROVED** [review]

File: ledger/complete/wal/checkpoint_v6_reader.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply ledger/complete/wal/checkpoint_v6_reader.go:174]: updated using Glob with `findCheckpointPartFiles`

**Pull request merged**